storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManager.java

```diff
@@ -646,70 +616,69 @@ public Optional<SnapshotFile> removeAndMarkSnapshotForDeletion(long snapshotOffs
     }
 
     public static List<ProducerStateEntry> readSnapshot(File file) throws IOException {
```

Comment:
It seems we don't have unit test for this method. We should add some tests for `readSnapshot` and `writeSnapshot` before introducing the generated code.

I have filed https://issues.apache.org/jira/browse/KAFKA-17135 for you @brandboat please feel free to un-assign if you have no bandwidth.

Reply:
Thank you 😃